thexpmap.cxx

```diff
@@ -964,8 +964,7 @@ void thexpmap::export_th2(class thdb2dprj * prj)
                 case TT_2DOBJ_PLACE_TOP: fprintf(pltf, " -place top"); break;
               }
               fprint_scale_option(pltf, pt);
-              if (pt->text) {
-                // attention: pt->text is reinterpret_cast<>'ed for some point types!
+              if (const auto* text = pt->get_text()) {
                 switch (pt->type) {
                   case TT_POINT_TYPE_SECTION:
                     fprintf(pltf, " -scrap %s", pt->station_name.print_name().c_str());
```

Comment:
is `case TT_POINT_TYPE_SECTION` still valid under `pt->get_text()` condition?

Reply:
Good question, I don't know. I'm trying to make the code more clear, there is only string used, so I check for the string. Maybe the condition should be inside the switch so we do not rule out `case TT_POINT_TYPE_SECTION`.

Reply:
This should be correct: https://github.com/speleo3/therion/commit/58433d5fddef1030a203d9b673d81c43902dbe1c

Reply:
Thanks, I have merged your changes.